### PR TITLE
[AGENT] kubernetes_cluster_id optimization

### PR DESCRIPTION
**Pull request by StrayLittlePunk**
### This PR is for:

- Agent

#### Changes to fix the bug
- When the kubernetes_cluster_id issued by the server is invalid, the agent obtains and updates the kubernetes_cluster_id

#### Affected branches
- main
#### Checklist
- [ ] Added unit test to verify the fix.

### CODE REVIEW

File: agent/src/trident.rs [hunk 1]
```diff
@@ -90,15 +90,22 @@ use crate::{
...
+    pub runtime_config: RuntimeConfig,
+    pub blacklist: Vec<u64>,
+    pub vm_mac_addrs: Vec<MacAddr>,
+    pub kubernetes_cluster_id: Option<String>,
+}
+
 pub enum State {
     Running,
-    ConfigChanged((RuntimeConfig, Vec<u64>, Vec<MacAddr>)), // (runtime_config, blacklist, vm_mac_addrs)
+    ConfigChanged(ChangedConfig), // (runtime_config, blacklist, vm_mac_addrs)
```
Comment (rvql) [inline agent/src/trident.rs:102]: 注释可以去掉了

File: agent/src/rpc/synchronizer.rs [hunk 2]
```diff
@@ -612,16 +614,24 @@ impl Synchronizer {
...
             tp::Status::Failed => warn!(
                 "trisolaris (ip: {}) responded with {:?}",
                 remote,
                 tp::Status::Failed
             ),
             tp::Status::Heartbeat => return,
+            tp::Status::ClusterIdNotFound => {
+                let k8s_cluster_id = Config::async_get_k8s_cluster_id(session).await;
+                kubernetes_cluster_id.replace(k8s_cluster_id.clone());
+                running_config.write().kubernetes_cluster_id = k8s_cluster_id;
```
Comment (sharang) [inline agent/src/rpc/synchronizer.rs:633]: 此时会打印日志吗？应该打印日志说明 cluster-id 变更了，并说明是因为控制器返回了 cluster-id-not-found 触发了 cluster-id 的重新获取

Comment (StrayLittlePunk) [reply agent/src/rpc/synchronizer.rs:633]: image

Comment (StrayLittlePunk) [reply agent/src/rpc/synchronizer.rs:633]: 在config/handler.rs 打印

Comment (StrayLittlePunk) [reply agent/src/rpc/synchronizer.rs:634]: 已经加上因为 cluster-id-not-found 触发了 cluster-id 的重新获取日志

**Review by rvql — APPROVED** [review]

File: agent/src/rpc/synchronizer.rs [hunk 3]
```diff
@@ -612,16 +614,25 @@ impl Synchronizer {
...
         match resp.status() {
             tp::Status::Failed => warn!(
                 "trisolaris (ip: {}) responded with {:?}",
                 remote,
                 tp::Status::Failed
             ),
             tp::Status::Heartbeat => return,
+            tp::Status::ClusterIdNotFound => {
+                let k8s_cluster_id = Config::async_get_k8s_cluster_id(session).await;
+                info!("get latest kubernetes_cluster_id={}, because of old kubernetes_cluster_id not found", k8s_cluster_id);
```
Comment (sharang) [inline agent/src/rpc/synchronizer.rs:632]: because the old kubernetes_cluster_id {} is no longer valid.

**Review by sharang — APPROVED** [review]

**Pull request merged**